### PR TITLE
8289006: Cleanup from thread.hpp split

### PR DESCRIPTION
**Pull request by coleenp**
Remove extra include directives.  Tested with GHA.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289006](https://bugs.openjdk.org/browse/JDK-8289006): Cleanup from thread.hpp split

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [f3e109c8](https://git.openjdk.org/jdk/pull/9252/files/f3e109c8f01b1b13f1bdc9859d5abfc41b21d6b1)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9252/head:pull/9252` \
`$ git checkout pull/9252`

Update a local copy of the PR: \
`$ git checkout pull/9252` \
`$ git pull https://git.openjdk.org/jdk pull/9252/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9252`

View PR using the GUI difftool: \
`$ git pr show -t 9252`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9252.diff">https://git.openjdk.org/jdk/pull/9252.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/zero/stack_zero.cpp [hunk 1]
```diff
@@ -26,11 +26,10 @@
 #include "precompiled.hpp"
 #include "interpreter/interpreterRuntime.hpp"
 #include "interpreter/zero/bytecodeInterpreter.hpp"
-#include "runtime/javaThread.hpp"
-#include "stack_zero.hpp"
-#include "stack_zero.inline.hpp"
 #include "runtime/frame.inline.hpp"
 #include "runtime/javaThread.inline.hpp"
+#include "stack_zero.hpp"
+#include "stack_zero.inline.hpp"
```
Comment (dholmes-ora) [inline src/hotspot/cpu/zero/stack_zero.cpp:32]: Existing, but it is redundant to include the .hpp and the .inline.hpp file as the later always includes the former.

**Review by dholmes-ora — APPROVED** [review]
One pre-existing nit but your changes look good.

Thanks.

File: src/hotspot/cpu/zero/stack_zero.cpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/cpu/zero/stack_zero.cpp]: You're right.  I touched that line but didn't even see it.  I confirmed stack_zero.inline.hpp includes stack_zero.hpp, so removed the latter from stack_zero.cpp.  We'll let GHA confirm.

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up. Thanks for taking care of my late comments.

**Pull request closed without merging**